src/rashdf/plan.py

```diff
@@ -1117,6 +1121,106 @@ def reference_lines_timeseries_output(self) -> xr.Dataset:
+        """
+
+        # Decode the contents of the DataFrame from utf-8
+        def decode_bytes(val):
+            if isinstance(val, bytes):
+                return val.decode("utf-8")
+            return val
+
+        if vartype == "Flow":
+            output_path = self.OBS_FLOW_OUTPUT_PATH
+        elif vartype == "Stage":
+            output_path = self.OBS_STAGE_OUTPUT_PATH
+        else:
+            raise ValueError('vartype must be either "Flow" or "Stage".')
+
+        observed_group = self.get(output_path)
+        if observed_group is None:
+            raise RasPlanHdfError(
+                f"Could not find HDF group at path '{output_path}'."
+                f" Does the Plan HDF file contain reference {vartype[:-1]} output data?"
+            )
+        print(f"observed_group: {observed_group}")
+        if "Attributes" in observed_group.keys():
+            attr_path = observed_group["Attributes"]
+            print(f"attr_path: {attr_path}")
+            attrs_df = pd.DataFrame(attr_path[:])
+            # Apply the decoding function to each element in the DataFrame
+            attrs_df = attrs_df.map(decode_bytes)
+            if vartype == "Flow":
+                attrs_df["Units"] = "cfs"
+            elif vartype == "Stage":
+                attrs_df["Units"] = "ft"
+            else:
+                attrs_df["Units"] = "Unknown"
+        das = {}
+        for site in observed_group.keys():
+            if site != "Attributes":
+                # Site Ex: 'Ref Point: Grapevine_Lake_RP'
+                site_path = observed_group[site]
+                prefix = site.split(":")[0]
+                suffix = site.split(":")[1][1:]
+                data_df = pd.DataFrame(site_path[:])
+                # Apply the decoding function to each element in the DataFrame
+                data_df = data_df.map(decode_bytes)
+                # Assign data types to the columns
+                data_df["Date"] = data_df["Date"].apply(parse_ras_datetime)
+                data_df["Date"] = pd.to_datetime(
+                    data_df["Date"], format="%d%b%Y %H:%M:%S"
+                )
+                data_df["Value"] = data_df["Value"].astype(float)
+                # Determine the site type
+                site_type = (
+                    "reference_line" if "Ref Line" in site else "reference_point"
+                )
+                # Package into an xarray DataArray
+                da = xr.DataArray(
+                    data_df["Value"].values,
+                    name=vartype,
+                    dims=["time"],
+                    coords={
+                        "time": data_df["Date"].values,
+                    },
+                    attrs={
+                        "units": attrs_df["Units"][0],
+                        "hdf_path": f"{output_path}/{site}",
+                    },
+                )
+                das[suffix] = da
+
+        return das
+
+    def observed_data_timeseries_input(self) -> xr.Dataset:
```

Comment:
Redundant method? Could this be removed?

Reply:
Method removed